cluster-autoscaler/core/scaledown/actuation/drain.go

```diff
@@ -224,6 +225,9 @@ func (e Evictor) initiateEviction(ctx *acontext.AutoscalingContext, node *apiv1.
 }
 
 func (e Evictor) evictPod(ctx *acontext.AutoscalingContext, podToEvict *apiv1.Pod, retryUntil time.Time, maxTermination int64, fullEvictionPod bool, force bool) status.PodEvictionResult {
+	if fake.IsFake(podToEvict) {
+		return status.PodEvictionResult{Pod: podToEvict, TimedOut: false, Err: nil}
```

Comment:
Why is returning a successful eviction result preferable to returning a "not found" error for a fake pod? Is this simply a runtime optimization that processes more cheaply through fake pods? If so, would it be better to add a fake pod filter to the `func podsToEvict` instead?